### PR TITLE
Improving the RequireJS documentation with another available config

### PR DESCRIPTION
**Pull request by eduard13**
## Purpose of this pull request

This pull request (PR) is improving the RequireJS documentation, by adding another available `text` config that can be used.

## Affected DevDocs pages

-  https://devdocs.magento.com/guides/v2.3/javascript-dev-guide/javascript/requirejs.html#requirejs-config-map

## Links to Magento source code

-  https://github.com/magento/magento2/blob/2.3/app/code/Magento/Theme/view/base/requirejs-config.js#L61

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: src/guides/v2.3/javascript-dev-guide/javascript/requirejs.md [hunk 1]
```diff
@@ -106,3 +107,19 @@ In this snippet, `Vendor_Module/js/module-mixin` will overwrite the existing com
 ```
 
 The concept of Javascript mixins itself is explained in depth in [Using Javascript Mixins]({{ page.baseurl }}/javascript-dev-guide/javascript/js_mixins.html).
+
+### text
+
+The `text` configuration is used to set the security request headers via the following JS file: [`text.js`]({{ site.mage2bloburl }}/{{ page.guide_version }}/lib/web/mage/requirejs/text.js)
```
Comment (jeff-matthews) [inline src/guides/v2.3/javascript-dev-guide/javascript/requirejs.md]:
```suggestion
The `text` configuration is used to set the security request headers using the [`text.js`]({{ site.mage2bloburl }}/{{ page.guide_version }}/lib/web/mage/requirejs/text.js) file.
```

File: src/guides/v2.3/javascript-dev-guide/javascript/requirejs.md [hunk 2]
```diff
@@ -106,3 +107,19 @@ In this snippet, `Vendor_Module/js/module-mixin` will overwrite the existing com
...
+
+Without [Cross Origin Resource Sharing (CORS)](https://en.wikipedia.org/wiki/Cross-origin_resource_sharing) it is not possible to add X-Requested-With header to a cross domain XHR request. So having this header set, the server knows that the request was initiated from the same domain.
```
Comment (jeff-matthews) [inline src/guides/v2.3/javascript-dev-guide/javascript/requirejs.md]:
```suggestion
Without [Cross Origin Resource Sharing (CORS)](https://en.wikipedia.org/wiki/Cross-origin_resource_sharing) it is not possible to add the `X-Requested-With` header to a cross domain XHR request. Set this header to tell the server that the request was initiated from the same domain.
```

**Comment by eduard13** [comment]
Updated. Thank you @jeff-matthews.

**Pull request merged**